modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C

```diff
@@ -1087,3 +1106,109 @@ CrackFrontDefinition::isNodeOnIntersectingBoundary(const Node * const node) cons
+    disp_current_node(1) = _subproblem.getVariable(_tid, _disp_y_var_name).getNodalValue(*current_node);
+    disp_current_node(2) = _subproblem.getVariable(_tid, _disp_z_var_name).getNodalValue(*current_node);
+
+    next_node = _mesh.nodePtr(_ordered_crack_front_nodes[1]);
+    disp_next_node(0) = _subproblem.getVariable(_tid, _disp_x_var_name).getNodalValue(*next_node);
+    disp_next_node(1) = _subproblem.getVariable(_tid, _disp_y_var_name).getNodalValue(*next_node);
+    disp_next_node(2) = _subproblem.getVariable(_tid, _disp_z_var_name).getNodalValue(*next_node);
+
+    //Calculate change in length of crack front edge and project in the tangent direction to get tangential strain
+    l1 = *next_node - *current_node;
+    l1 = (l1 * _tangent_directions[0]) * _tangent_directions[0];
+    delta_l1 = disp_next_node - disp_current_node;
+    delta_l1 = (delta_l1 * _tangent_directions[0]) * _tangent_directions[0];
+    _strain_along_front[0] = delta_l1.size() / l1.size();
+  }
+
+  for (unsigned int i=1; i<num_crack_front_nodes-1; ++i)
+  {
+    current_node = _mesh.nodePtr(_ordered_crack_front_nodes[i]);
+    if (current_node->processor_id() == processor_id())
+    {
+      disp_current_node(0) = _subproblem.getVariable(_tid, _disp_x_var_name).getNodalValue(*current_node);
+      disp_current_node(1) = _subproblem.getVariable(_tid, _disp_y_var_name).getNodalValue(*current_node);
+      disp_current_node(2) = _subproblem.getVariable(_tid, _disp_z_var_name).getNodalValue(*current_node);
+
+      previous_node = _mesh.nodePtr(_ordered_crack_front_nodes[i-1]);
+      disp_previous_node(0) = _subproblem.getVariable(_tid, _disp_x_var_name).getNodalValue(*previous_node);
+      disp_previous_node(1) = _subproblem.getVariable(_tid, _disp_y_var_name).getNodalValue(*previous_node);
+      disp_previous_node(2) = _subproblem.getVariable(_tid, _disp_z_var_name).getNodalValue(*previous_node);
+
+      next_node = _mesh.nodePtr(_ordered_crack_front_nodes[i+1]);
+      disp_next_node(0) = _subproblem.getVariable(_tid, _disp_x_var_name).getNodalValue(*next_node);
+      disp_next_node(1) = _subproblem.getVariable(_tid, _disp_y_var_name).getNodalValue(*next_node);
+      disp_next_node(2) = _subproblem.getVariable(_tid, _disp_z_var_name).getNodalValue(*next_node);
+
+      l0 = *current_node - *previous_node;
+      l0 = (l0 * _tangent_directions[i]) * _tangent_directions[i];
+      delta_l0 = disp_current_node - disp_previous_node;
+      delta_l0 = (delta_l0 * _tangent_directions[i]) * _tangent_directions[i];
+      l1 = *next_node - *current_node;
+      l1 = (l1 * _tangent_directions[i]) * _tangent_directions[i];
+      delta_l1 = disp_next_node - disp_current_node;
+      delta_l1 = (delta_l1 * _tangent_directions[i]) * _tangent_directions[i];
+      _strain_along_front[i] = 0.5 * ( delta_l0.size()/l0.size() + delta_l1.size()/l1.size() );
+    }
+  }
+
+  current_node = _mesh.nodePtr(_ordered_crack_front_nodes[num_crack_front_nodes-1]);
```

Comment:
Really good place for a MooseAssert(). The nodePtr() method will return NULL if it can't find a node by number. If you're sure it won't happen an assert is good, but if there's a chance you might want an actual test if to verify that it's good.